src/package-operations.test.ts

```diff
@@ -128,7 +120,6 @@ describe('package-operations', () => {
   describe('getMetadataForAllPackages', () => {
     const readdirMock = jest.spyOn(fs.promises, 'readdir');
 
-    let mockIndex = -1;
```

Comment:
This was... an interesting mock! I think I was able to accomplish the same goal by storing the mockIndex in a closure that is created per-test, rather than in the parent scope. See the `getMockReadJsonFile` function.

Let me know if I've misunderstood the intent though! I assumed it was that each time `readJsonFile` was called, that it would return the next name.